asyncscope.md

```diff
@@ -607,6 +608,78 @@ class Camera {
 }
 ```
 
+## Recursively spawning work until completion
+Below are two ways you could recursively spawn work on a scope using `let_with_async_scope` or `counting_scope`. Both
+examples are perfectly valid ways to ensure that all new work spawned from previous work will complete.
+
+:::cmptable
+
+### let_with_async_scope
+```cpp
+struct tree {
+  std::unique_ptr<tree> left;
+  std::unique_ptr<tree> right;
+  int data;
+};
+
+auto process(ex::scheduler auto sch, ex::counting_scope& scope, tree& t) {
+  return ex::let_error(ex::schedule(sch) | then([sch, &]() noexcept {
+    if (t.left)
+      ex::spawn(scope, process(sch, scope, t.left.get()));
+    if (t.right)
+      ex::spawn(scope, process(sch, scope, t.right.get()));
+    do_stuff(t.data);
+  }), | ex::let_error([](auto& e) {
+    // log error
+    return just(-1);
+  });
+}
+
+int main() {
+  ex::scheduler sch;
+  tree t = make_tree();
+  // let_with_async_scope will ensure all new work will be spawned on the
+  // scope and will not be joined until all work is finished
+  this_thread::sync_wait(ex::let_with_async_scope([&, sch](auto scope) {
+	return process(sch, scope, t);
+  }));
+}
+```
+
+### counting_scope
+```cpp
+struct tree {
+  std::unique_ptr<tree> left;
+  std::unique_ptr<tree> right;
+  int data;
+};
+
+auto process(ex::scheduler auto sch, tree& t) {
+  return ex::let_value_with([]() noexcept { return ex::counting_scope{}; }, [&](ex::counting_scope& scope) noexcept {
+	return ex::let_error(ex::schedule(sch) | ex::then([sch, &]() noexcept {
+    	    if (t.left)
+      	        ex::spawn(scope, process(sch, t.left.get()));
+	    if (t.right)
+                ex::spawn(scope, process(sch, t.right.get()));
+		do_stuff(t.data);
+  	}), [](auto& e) {
+	    // log error
+            return just(-1);
+        }) | let_value([&scope]() noexcept {
+	    return scope.join();
+	});
+  });
```

Comment:
`let_value_with` is a Unifex algorithm, but it hasn't been proposed for C++26. This example also has the same problem with `just(-1)`.
```suggestion
  return unifex::let_value_with([]() noexcept { return ex::counting_scope{}; },
    [&](ex::counting_scope& scope) noexcept {
	return ex::let_error(ex::schedule(sch) | ex::then([sch, &]() noexcept {
    	    if (t.left)
      	        ex::spawn(scope, process(sch, t.left.get()));
	    if (t.right)
                ex::spawn(scope, process(sch, t.right.get()));
            do_stuff(t.data);
  	}), [](auto& e) {
	    // log error
            return ex::just();
        }) | ex::let_value([&scope]() noexcept {
	    return scope.join();
	});
  });
```